lib/appmap/service/config_analyzer.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module AppMap
+  module Service
+    class ConfigAnalyzer
+      attr_reader :config_error
+
+      def initialize(config_file)
+        @config_file = config_file
+        @config = load_config
+      end
+
+      def app_name
+        @config.to_h[:name] if present?
+      end
+
+      def present?
+        File.exist?(@config_file)
+      end
+
+      def valid?
+        present? && @config.to_h.key?(:name) && @config.to_h.key?(:packages)
```

Comment:
Isn't it the case that `present?` can be true but `@config` is nil?

Reply:
Should not be a problem as `nil.to_h` gives an empty hash:
```
irb(main):001:0> nil.to_h
=> {}
```